speech/cloud-client/beta_snippets.py

```diff
@@ -153,30 +152,42 @@ def transcribe_file_with_diarization(speech_file):
 
     config = speech.types.RecognitionConfig(
         encoding=speech.enums.RecognitionConfig.AudioEncoding.LINEAR16,
+        sample_rate_hertz=8000,
         language_code='en-US',
         enable_speaker_diarization=True,
         diarization_speaker_count=2)
 
     print('Waiting for operation to complete...')
     response = client.recognize(config, audio)
 
+    # response.results contains partial results with the last item
+    # containing the entire result:
     result = response.results[-1]
+
     words_info = result.alternatives[0].words
-    pieces = ['%s (%s)' % (word_info.word, word_info.speaker_tag)
-              for word_info in words_info]
-    print(' '.join(pieces))
+
+    # Separating the words by who said what:
+    speakers_words = []
+    for word_info in words_info:
+        if speakers_words and speakers_words[-1][0] == word_info.speaker_tag:
```

Comment:
If I understand correctly what this piece of for loop is doing (creating list of words for tag) isn't is better to use hashmap. Does this loop work what it's supposed to do?

Reply:
Jerjou is already reviewing this code so I should stop reviewing. You are already is great hands :)